lib/pure/collections/deques.nim

```diff
@@ -274,8 +274,9 @@ proc expandIfNeeded[T](deq: var Deque[T]) =
   if unlikely(deq.count >= cap):
     var n = newSeq[T](cap * 2)
     var i = 0
-    for x in mitems(deq): # don't use copyMem because of the GC and because it's slower.
-      n[i] = move(x)
+    for x in mitems(deq):
+      when nimVM: n[i] = x # workaround for VM bug
+      else: n[i] = move(x)
       inc i
```

Comment:
Could this be changed to
```
for i, x in mpairs(deq):
  when nimVM: n[i] = x
  else: n[i] = move(x)
```
?
Curious, which VM bug is it?

Reply:
Probably but not important enough. Plus I don't really like the ``mpairs`` / ``pairs`` design for seqs and arrays.